jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/MultiPartFormInputStream.java

```diff
@@ -809,6 +825,9 @@ public boolean content(ByteBuffer buffer, boolean last)
         public void startPart()
         {
             reset();
+            _numParts++;
+            if (_numParts >= _maxParts)
```

Comment:
`_maxParts` can be set to a negative value, so I'd change to `if (_maxParts >= 0 && _numParts >= _maxParts)`.

Reply:
done